describe validationData, authorizer (#9)
eth-infinitism/RIPs: RIPS/rip-7560.md

The frame must return 32 bytes `validationData` that is interpreted as:

> - **authorizer** (20 bytes)

Author: ```suggestion
- **authorizer** (20 bytes) - (magic value/address, defined below)
```

> - **authorizer** (20 bytes) - (magic value/address, defined below)

Reviewer: Don't introduce the new term "authorizer" in this EIP. Just make it 20 bytes. We can add the explanation in the "Rationale" section.

Author: then how to call it? 20 byte magic?

- **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
- **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.

Currently, **authorizer** is treated as a magic value:

> - `MAGIC_VALUE_SENDER` - returned by the account after it verified the transaction (and agrees to pay for its execution)

Reviewer: Remove the brackets "(" and ")". They clutter the text and add no value.

Author: not sure to which brakets you mean to remove.

> - `MAGIC_VALUE_SIGFAIL` - returned by the account if the transaction is syntactically valid, but the signature is not.

Author: ```suggestion
- `MAGIC_VALUE_SIGFAIL` - This value is returned by the account if the transaction is syntactically valid, but the signer is incorrect.
```

>     This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.

Reviewer: This *value*

> - Any other returned value is considered failure, and the transaction is rejected.

Reviewer: ... considered **to be a** failure

> - Any other returned value is considered failure, and the transaction is rejected.

Author: ```suggestion
- Any other returned value is considered to be a failure, and the transaction is rejected.
```

> - Future protocol MAY define authorizer as an on-chain contract to perform further account authorization, such as aggregation

Reviewer: Move this to rationale

> A revert by `validateTransaction` indicates rejection of the transaction, and it is not included on-chain.

Reviewer: Revert and invalid return value are equivalent, so they should be in the same element in the list.

#### Paymaster validation frame

The `paymaster` of the transaction, if specified, is invoked with the following data:

```solidity

function validatePaymasterTransaction(uint256 version, bytes32 txHash, bytes transaction) 
external returns (bytes32 validationData, bytes context);

```

The gas limit of this frame is set to `paymasterValidationGasLimit`.

The amount of gas used by this frame is referred to as `paymasterValidationGasUsed`.

> - The `version` parameter is `AA_TX_TYPE`

Reviewer: No, it isn't. The `version` is a mechanism to change the structure without necessarily changing the transaction type. This mechanism is still TBD though.

Author: then suggest a fix to this line - either remove it altogether, or suggest another value (e.g. AA_TX_VERSION?)

- The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.\
- The `txHash` parameter represents the hash of the AA_TX_TYPE transaction with empty signature, as defined in section
[Calculation of Transaction Type AA_TX_TYPE hash](#calculation-of-transaction-type-aatxtype-hash).

The frame returns 

> - `validationData` encoded as:

Reviewer 2: This is a duplicate, if `validationData` is identical, we can refer to the previous definition

> - `validationData` encoded as:

Author: ```suggestion
- `validationData` - encoded the same as described above, in validateTransaction.
```

    - **authorizer** (20 bytes)
    - **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
    - **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
- `context` - optional buffer returned by the paymaster, which is later passed to `postPaymasterTransaction`

>     This value MUST NOT exceed `MAX_CONTEXT_SIZE`.

Reviewer: Let's define `MAX_CONTEXT_SIZE` somewhere?

Author: defined in the constants section

The  **authorizer** is treated as a magic value:
- `MAGIC_VALUE_PAYMASTER` - returned by the paymaster after it verified the transaction (and agrees to pay for its execution)
- `MAGIC_VALUE_SIGFAIL` - returned by the paymaster if its paymasterData is expected to contain a signature, but its current value is not valid.

>     This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.

Reviewer: This **value**

>     This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.

Author: ```suggestion
    This value is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.
```

[Gas fees charged for transaction input](#gas-fees-charged-for-transaction-input) section.\
The amount of gas used by this frame is referred to as `gasUsedByExecution`.

> The transaction is included in the block even if the execution frame reverts.

Reviewer: I think my version is more explicit.

> The transaction is included in the block even if the execution frame reverts.

Author: ```suggestion
The validation frames do not revert even if the execution frame reverts.
```
